nucliadb/nucliadb/learning_collector.py

```diff
@@ -0,0 +1,158 @@
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+import contextlib
+import json
+import logging
+from collections.abc import AsyncIterator
+from typing import Any, Optional, Type, Union
+
+import httpx
+from fastapi import Request, Response
+from fastapi.responses import StreamingResponse
+
+from nucliadb_telemetry import errors
+from nucliadb_utils.settings import nuclia_settings
+
+SERVICE_NAME = "nucliadb.learning_config"
+logger = logging.getLogger(SERVICE_NAME)
+
+
+NUCLIA_ONPREM_AUTH_HEADER = "X-NUCLIA-NUAKEY"
+
+
+async def proxy(
+    request: Request, method: str, url: str, headers: Optional[dict[str, str]] = None
+) -> Union[Response, StreamingResponse]:
+    """
+    Proxy the request to the learning collector API.
+
+    request: Request. The incoming request.
+    method: str. The HTTP method to use.
+    url: str. The URL to proxy the request to.
+
+    Returns: Response. The response from the learning collector API.
+    If the response is chunked, a StreamingResponse is returned.
+    """
+    proxied_headers = headers or {}
+    proxied_headers.update(request.headers)
+    async with learning_collector_client() as client:
+        try:
+            response = await client.request(
+                method=method.upper(),
+                url=url,
+                params=request.query_params,
+                content=await request.body(),
+                headers=proxied_headers,
+            )
+        except Exception as exc:
+            errors.capture_exception(exc)
+            return Response(
+                content=b"Unexpected error while trying to proxy the request to the learning collector API. Please try again later.",  # noqa
+                status_code=503,
+                media_type="text/plain",
+            )
+        if response.headers.get("Transfer-Encoding") == "chunked":
+            return StreamingResponse(
+                content=response.aiter_bytes(),
+                status_code=response.status_code,
+                headers=response.headers,
+                media_type=response.headers.get("Content-Type"),
+            )
+        return Response(
+            content=response.content,
+            status_code=response.status_code,
+            headers=response.headers,
+            media_type=response.headers.get("Content-Type"),
+        )
+
+
+def is_onprem() -> bool:
+    return nuclia_settings.nuclia_service_account is not None
```

Comment:
I think we should be using `nucliadb_utils.settings` `NucliaSettings.onprem` instead, as it's the one used in other places